### PR TITLE
feat: db model types migration

### PR DESCRIPTION
**Pull request by ayushjain17**
## Problem
Model types are very tightly tied down to cac and exp crates via diesel, due to which usage of these types in any other crate requires redefining the type again in that crate

## Solution
Move out the migrations to separate crates (`cac_db_config` and `experimentation_db_config`) for migrations only, which include the corresponding schema.rs file
using this schema.rs defined the models in `superposition_types` which conditionally implement the diesel traits, when `diesel_derives` feature is enabled

## Note
The change-log seems big, because the migration files (sql files) where moved along with few other files which increased the number of effected files

### This PR is not complete, it only introduces the separation, the usage of the types will be raised a separate PR

### CODE REVIEW

File: crates/cac_db_config/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "cac_db_config"
```
Comment (ShubhranshuSanjeev) [inline crates/cac_db_config/Cargo.toml:2]: The package name can be just `cac_db`

Comment (ShubhranshuSanjeev) [reply crates/cac_db_config/Cargo.toml:2]: Same for `experimentation_db_config` -> `experimentation_db`

File: crates/superposition_types/Cargo.toml [hunk 2]
```diff
@@ -8,23 +8,34 @@ edition = "2021"
...
 
 [features]
 default = ["server"]
-diesel_derives = ["dep:diesel", "dep:superposition_derives"]
+diesel_derives = [
+    "dep:diesel",
+    "dep:superposition_derives",
+    "dep:cac_db_config",
+    "cac_db_config/diesel_derives",
+    "experimentation_db_config?/diesel_derives",
```
Comment (ShubhranshuSanjeev) [inline crates/superposition_types/Cargo.toml:33]: What does `?` do here?

File: crates/superposition_types/src/cac_models.rs
Comment (ShubhranshuSanjeev) [inline crates/superposition_types/src/cac_models.rs:1]: The models can structured as such `models/cac.rs` and `modes/exp.rs`

**Review by ShubhranshuSanjeev — CHANGES_REQUESTED** [review]

File: crates/superposition_types/Cargo.toml (hunk 2 above)
Comment (ayushjain17) [reply crates/superposition_types/Cargo.toml]: it is used while enabling certain feature of another dependency which in turn in optional and is dependent on some other feature to enable the usage of that dependency
check the section before the second Note [here](https://doc.rust-lang.org/cargo/reference/features.html#dependency-features)

**Review by Datron — APPROVED** [review]

**Review by Datron — APPROVED** [review]

**Pull request merged**